rust/operator-binary/src/command.rs

```diff
@@ -0,0 +1,170 @@
+use stackable_trino_crd::{
+    TrinoCluster, CONFIG_DIR_NAME, DATA_DIR_NAME, ENV_S3_ACCESS_KEY, ENV_S3_SECRET_KEY,
+    HIVE_PROPERTIES, PASSWORD_DB, RW_CONFIG_DIR_NAME, S3_SECRET_DIR_NAME, SECRET_KEY_S3_ACCESS_KEY,
+    SECRET_KEY_S3_SECRET_KEY, STACKABLE_CLIENT_TLS_DIR, STACKABLE_INTERNAL_TLS_DIR,
+    STACKABLE_MOUNT_CLIENT_TLS_DIR, STACKABLE_MOUNT_INTERNAL_TLS_DIR, STACKABLE_TLS_STORE_PASSWORD,
+    SYSTEM_TRUST_STORE, SYSTEM_TRUST_STORE_PASSWORD, USER_PASSWORD_DATA_DIR_NAME,
+};
+
+pub fn container_prepare_args(
+    trino: &TrinoCluster,
+    s3_spec: Option<&S3ConnectionSpec>,
+) -> Vec<String> {
+    let mut args = vec![
+        // Copy system truststore to stackable client tls dir
+        format!("keytool -importkeystore -srckeystore {SYSTEM_TRUST_STORE} -srcstoretype jks -srcstorepass {SYSTEM_TRUST_STORE_PASSWORD} -destkeystore {STACKABLE_CLIENT_TLS_DIR}/truststore.p12 -deststoretype pkcs12 -deststorepass {STACKABLE_TLS_STORE_PASSWORD} -noprompt"),
+    ];
+
+    // User password data
+    if trino.get_authentication().is_some() {
+        args.extend(create_key_and_trust_store(
+            STACKABLE_MOUNT_CLIENT_TLS_DIR,
+            STACKABLE_CLIENT_TLS_DIR,
+            "stackable-client-ca-cert",
+        ));
+        args.extend(chown_and_chmod(USER_PASSWORD_DATA_DIR_NAME));
+    } else if trino.get_client_tls().is_some() {
+        args.extend(create_key_and_trust_store(
+            STACKABLE_MOUNT_CLIENT_TLS_DIR,
+            STACKABLE_CLIENT_TLS_DIR,
+            "stackable-client-ca-cert",
+        ));
+    }
+
+    // Chown and mod the certificates dir (this will always be created even if no TLS is required)
+    args.extend(chown_and_chmod(STACKABLE_CLIENT_TLS_DIR));
+
+    if trino.get_internal_tls().is_some() {
+        args.extend(create_key_and_trust_store(
+            STACKABLE_MOUNT_INTERNAL_TLS_DIR,
+            STACKABLE_INTERNAL_TLS_DIR,
+            "stackable-internal-ca-cert",
+        ));
+        args.extend(chown_and_chmod(STACKABLE_INTERNAL_TLS_DIR));
+        // add cert to internal truststore
+        if trino.tls_enabled() {
+            args.push(format!("keytool -importcert -file {STACKABLE_MOUNT_CLIENT_TLS_DIR}/ca.crt -alias stackable-ca-cert -keystore {STACKABLE_INTERNAL_TLS_DIR}/truststore.p12 -storepass {STACKABLE_TLS_STORE_PASSWORD} -noprompt"));
+        }
+    }
+
+    // Load S3 ca to truststore if S3 TLS enabled
+    if let Some(s3) = s3_spec {
+        if let Some(Tls {
+            verification:
+                TlsVerification::Server(TlsServerVerification {
+                    ca_cert: CaCert::SecretClass(secret_class),
+                }),
+        }) = &s3.tls
+        {
+            args.push(format!("keytool -importcert -file {STACKABLE_CLIENT_TLS_DIR}/{s3_certs_dir}/ca.crt -alias stackable-{secret_class} -keystore {STACKABLE_CLIENT_TLS_DIR}/truststore.p12 -storepass {STACKABLE_TLS_STORE_PASSWORD} -noprompt", s3_certs_dir = secret_class));
```

Comment:
Aren't the trusted-server and trusted-client truststores separate?

Reply:
I did not find anything like that. For client / server the same properties / directories are used.

Reply:
From what I can tell, you specify the HTTPS server's trust store with `http-server.https.truststore.path`, the internal communications trust store with `internal-communication.https.truststore.path`, and the HTTPS client's trust store with `javax.net.ssl.trustStore` (though separately, in jvm.config).

Reply:
split into client, server, internal truststores